homeassistant/components/smarty/__init__.py

```diff
@@ -0,0 +1,65 @@
+"""Support to control a Salda Smarty XP/XV ventilation unit."""
+
+from datetime import timedelta
+import ipaddress
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (CONF_NAME, CONF_HOST)
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['pysmarty==0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'smarty'
+DATA_SMARTY = 'smarty'
+SMARTY_NAME = 'Smarty'
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema({
+            vol.Required(CONF_HOST): vol.All(ipaddress.ip_address, cv.string),
+            vol.Optional(CONF_NAME, default=SMARTY_NAME): cv.string
+        }),
+    },
+    extra=vol.ALLOW_EXTRA)
+
+RPM = 'rpm'
+SIGNAL_UPDATE_SMARTY = 'smarty_update'
+
+
+def setup(hass, config):
+    """Set up the smarty environment."""
+    from pysmarty import (Smarty)
+    conf = config[DOMAIN]
+
+    host = conf.get(CONF_HOST)
+    name = conf.get(CONF_NAME)
+
+    _LOGGER.debug("Name: %s, host: %s", name, host)
+
+    smarty = Smarty(host=host, loop=hass.loop)
+    hass.data[DATA_SMARTY] = smarty
+    hass.data[SMARTY_NAME] = name
```

Comment:
Only use the `DOMAIN` as the key for this integration in `hass.data` and put everything else inside there, like:
```python
hass.data[DOMAIN] = {
    'api': smarty,
    'name': name
}